fs/dafilesrv/dafilesrv.cpp

```diff
@@ -386,13 +386,119 @@ int main(int argc,char **argv)
+        }
+        else if ((argc>i+1)&&(stricmp(argv[i],"-I")==0)) {
+            i++;
+            componentName.clear().append(argv[i++]);
+        }
+        else if ((argc>i+1)&&(stricmp(argv[i],"-p")==0)) {
+            i++;
+            listenep.port = atoi(argv[i++]);
+        }
+        else if ((argc>i+1)&&(stricmp(argv[i],"-addr")==0)) {
+            i++;
+            if (strchr(argv[i],'.')||!isdigit(argv[i][0]))
+                listenep.set(argv[i], listenep.port);
+            else
+                listenep.port = atoi(argv[i]);
+            i++;
+        }
+        else if ((argc>i+1)&&(stricmp(argv[i],"-sslp")==0)) {
+            i++;
+            sslport = atoi(argv[i++]);
+        }
+        else if ((argc>i+1)&&(stricmp(argv[i],"-sbsize")==0)) {
+            i++;
+            sendbufsize = atoi(argv[i++]);
+        }
+        else if ((argc>i+1)&&(stricmp(argv[i],"-rbsize")==0)) {
+            i++;
+            recvbufsize = atoi(argv[i++]);
+        }
+        else if (stricmp(argv[i],"-h")==0) {
+            usage();
+            exit(0);
+        }
+        else if (stricmp(argv[i],"-LOCAL")==0) {
+            i++;
+            locallisten = true;
+        }
+        else if (stricmp(argv[i],"-NOSSL")==0) { // overrides config setting
+            i++;
+            if (connectMethod == SSLOnly || connectMethod == SSLFirst || connectMethod == UnsecureFirst)
+            {
+                PROGLOG("DaFileSrv SSL specified in config but overridden by -NOSSL in command line");
+                connectMethod = SSLNone;
+            }
+        }
+        else
+            break;
+    }
+
+#ifdef _WIN32
+    if ((argc>i)&&(stricmp(argv[i],"-install")==0)) {
+        if (installService(DAFS_SERVICE_NAME,DAFS_SERVICE_DISPLAY_NAME,NULL)) {
+            PROGLOG(DAFS_SERVICE_DISPLAY_NAME " Installed");
+            return 0;
+        }
+        return 1;
+    }
+    if ((argc>i)&&(stricmp(argv[i],"-remove")==0)) {
+        if (uninstallService(DAFS_SERVICE_NAME,DAFS_SERVICE_DISPLAY_NAME)) {
+            PROGLOG(DAFS_SERVICE_DISPLAY_NAME " Uninstalled");
+            return 0;
+        }
+        return 1;
+    }
+#endif
+    if (argc > i) {
+        if (strchr(argv[i],'.')||!isdigit(argv[i][0]))
+            listenep.set(argv[i], listenep.port);
+        else
+            listenep.port = atoi(argv[i]);
+        sendbufsize = (argc>i+1)?(atoi(argv[i+1])*1024):0;
+        recvbufsize = (argc>i+2)?(atoi(argv[i+2])*1024):0;
+    }
+
     Owned<IPropertyTree> env = getHPCCEnvironment();
     IPropertyTree *keyPairInfo = nullptr;
     if (env)
     {
         StringBuffer dafilesrvPath("Software/DafilesrvProcess");
-        if (instanceName.length())
-            dafilesrvPath.appendf("[@name=\"%s\"]", instanceName.str());
+        if (componentName.length())
+            dafilesrvPath.appendf("[@name=\"%s\"]", componentName.str());
+        else
+            dafilesrvPath.append("[1]"); // in absence of name, use 1st
```

Comment:
Is this the primary/only functional change in this PR ? (the other changes being general refactoring)

Reply:
@mckellyln - no, the main change is moving the command line argument parsing before the above happens.
Previously it was later and therefore componentName was not set (it is passed on the command line)